### PR TITLE
Updated the cleaning code

### PR DESCRIPTION
**Pull request by adambushman**
## Description

Turns out our logic for processing the "test" data removed records (makes sense since we implemented that logic for the training data and we used the same function). That means our submission file had too few predicted records. 

The adjustments I made were three fold:

* Added argument to the function indicating if the data to process was "training" or "testing
* Modified logic for intentionally omiting select values (i.e. `OWN_XCAR_AGE`) to use median if there were missing values
* Added logic at the end to only omit remaining NAs if it was "training"; otherwise, impute the value per this logic: 1) if `logical` or `character`, take a random value based on the distribution, 2) if `numeric`, take the median of non-NAs

### New `test` data

The first thing to do is generate updated `application_train_clean`. Follow these steps:

1. Open `Data Cleaning.Rmd` **[DO NOT "Run All"]**
2. Navigate to lines 149-160
3. Click "Run Above"
4. After processing, select "Run Chunk"
5. Close `Data Cleaning.Rmd`; **[DO NOT run the smote section]**

Optionally, just run the pieces related to the `train` data (though doing the above won't take very long and won't hurt anything).

### Updated Final Model Predictions

The next thing to do is to open and run `Random Forest - Test Set.Rmd`; probably want to render it to HTML too so all that looks correct. Confirm the number of rows in the resulting predictions are `48744`.

### CODE REVIEW

**Comment by adambushman** [comment]
Fixed the cleaning code. Turns out there was a `OWN_xCAR_AGE` reference instead of the real `OWN_CAR_AGE`. I must have "fat-fingered" at some point. I thoroughly tested and we should be good to go.

**Review by aspendlo92 — APPROVED** [review]
Approved
